src/dist/replication/meta_server/meta_service.cpp

```diff
@@ -231,29 +231,33 @@ void meta_service::start_load_balance()
     _started = true;
 }
 
-bool meta_service::check_primary(dsn_message_t req)
+int meta_service::check_primary(dsn_message_t req)
 {
     if (!_failure_detector->is_primary())
     {
+        dsn_msg_options_t options;
+        dsn_msg_get_options(req, &options);
+        if ( options.context.u.is_forward_disabled )
```

Comment:
Is this configuration (as well as the above xxx_in_one_piece) for one single message? or for one kind of rpc message (e.g., for all RPC_PREPARE messages)? If the latter case is true, we should remove these flags, and make them available in task_spec  (i.e., configurable in config.ini). For the former case, would you please illustrate some cases so we can do assessment about why this is necessary?